File: src/main/java/dev/tricht/lunaris/listeners/GameEvent.java
package dev.tricht.lunaris.listeners;

import dev.tricht.lunaris.item.Item;
import lombok.Data;
import org.jnativehook.keyboard.NativeKeyEvent;

import java.awt.*;

@Data
public class GameEvent {

    private Item item;
    private Point mousePos;
    private int mouseWheelRotation;
    private NativeKeyEvent originalEvent;
}

File: src/main/java/dev/tricht/lunaris/listeners/GameEvent.kt
package dev.tricht.lunaris.listeners

import dev.tricht.lunaris.item.Item
import org.jnativehook.keyboard.NativeKeyEvent
import java.awt.Point

data class GameEvent(var item: Item? = null,
                     var mousePos: Point? = null,
                     var mouseWheelRotation: Int = 0,
                     var originalEvent: NativeKeyEvent? = null
)